rust/frontend/src/server.rs

```diff
@@ -551,26 +554,69 @@ async fn create_tenant(
 )]
 async fn get_tenant(
     headers: HeaderMap,
-    Path(name): Path<String>,
+    Path(tenant): Path<String>,
     State(mut server): State<FrontendServer>,
 ) -> Result<Json<GetTenantResponse>, ServerError> {
     server.metrics.get_tenant.add(1, &[]);
-    tracing::info!(name: "get_tenant", tenant_name = %name);
+    tracing::info!(name: "get_tenant", tenant = %tenant);
     server
         .authenticate_and_authorize(
             &headers,
             AuthzAction::GetTenant,
             AuthzResource {
-                tenant: Some(name.clone()),
+                tenant: Some(tenant.clone()),
                 database: None,
                 collection: None,
             },
         )
         .await?;
-    let request = GetTenantRequest::try_new(name)?;
+    let request = GetTenantRequest::try_new(tenant)?;
     Ok(Json(server.frontend.get_tenant(request).await?))
 }
 
+#[derive(Deserialize, Serialize, ToSchema, Debug)]
+pub struct UpdateTenantPayload {
+    pub resource_name: String,
+}
+
+/// Updates an existing tenant by ID.
+#[utoipa::path(
+    patch,
+    path = "/api/v2/tenants/{tenant}",
+    params(
+        ("tenant" = String, Path, description = "ID of the tenant to update")
+    ),
+    request_body = UpdateTenantPayload,
```

Comment:
Shouldn't there be an error case here for trying to update when its already set, since its immutable?

Reply:
Now returns `409`